test/spec/memory_fill.txt

```diff
@@ -10,7 +10,7 @@ out/test/spec/memory_fill.wast:118: assert_trap passed: out of bounds memory acc
 test() =>
 test() =>
 out/test/spec/memory_fill.wast:175: assert_invalid passed:
-  out/test/spec/memory_fill/memory_fill.8.wasm:000002c: error: memory variable out of range: 0 (max 0)
+  0000000: error: memory variable out of range: 0 (max 0)
```

Comment:
Why is this zero?

Reply:
It's because of the change here: https://github.com/fix-project/wabt/blob/simd-mm-fix/src/shared-validator.cc#L922

Previously the SharedValidator was hardcoding `Var(0, loc)` as the memidx, so it had the current location in the binary module. Now it's just using the memidx it gets from the IR, which came from an Index so didn't have a location attached to it.

Reply:
This seems like a regression, no?   It looks like the `Location` is passed to OnMemoryFill so why not use that?   ...  can't CheckMemoryIndex take advantage of it?  Or alternatively, why doesn't the incoming `Var memidx` have location associated with it?

Reply:
Basically the BinaryReaderIR and BinaryReaderInterp were converting a lot of Index to Vars (including memidx's) without including the location. Fixed in #1963 which I split out from this one because it includes a lot of changing test output.